### PR TITLE
Error handling for invalid request tokens

### PR DESCRIPTION
**Pull request by medhabansal**

### CODE REVIEW

File: lib/omniauth/strategies/mediawiki.rb [hunk 1]
```diff
@@ -115,6 +124,10 @@ def authorize_url_parameters
         end
       end
 
+      def valid_request_token?(request_token)
+        request_token.callback_confirmed? && request_token.token && request_token.secret
```
Comment (ragesoss) [inline lib/omniauth/strategies/mediawiki.rb]: I'm not sure what `callback_confirmed?` is for. Are you sure this being truthy is required for a valid request?

File: lib/omniauth/strategies/mediawiki.rb [hunk 2]
```diff
@@ -43,8 +47,13 @@ def request_phase
         end
 
         r.finish
+      rescue RequestTokenError => e
+        puts "There was a problem in getting the token from Mediawiki, here's what Mediawiki returned:"
+        puts request_token.inspect
```
Comment (ragesoss) [inline lib/omniauth/strategies/mediawiki.rb]: This is going to be a fatal NoMethodError somewhere else if this returns nil. We may as well re-raise the error after printing the debug info.

**Comment by ragesoss** [comment]
What problem are you having with this? It seems like it's working, although I haven't been able to replicate the timestamp problem.

File: lib/omniauth/strategies/mediawiki.rb (hunk 1 above)
Comment (medhabansal) [reply lib/omniauth/strategies/mediawiki.rb]: @ragesoss It was being returned as false for all invalid requests.

File: lib/omniauth/strategies/mediawiki.rb [hunk 3]
```diff
@@ -41,10 +45,15 @@ def request_phase
                        oauth_consumer_key: consumer.key
           ))
         end
-
         r.finish
+      rescue RequestTokenError => e
+        puts "There was a problem in getting the token from Mediawiki, here's what Mediawiki returned:"
+        puts request_token.inspect
+        raise RequestTokenError
```
Comment (ragesoss) [inline lib/omniauth/strategies/mediawiki.rb]: We should `raise e` here rather raising a new instance of RequestTokenError.

**Pull request merged**